datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1534,10 +1537,10 @@ mod tests {
         let expr = cast_to_int64_expr(now_expr()) + lit(100_i64);
         test_evaluate_with_start_time(expr, lit(ts_nanos + 100), &time);
 
-        //  CAST(now() as int64) < cast(to_timestamp(...) as int64) + 50000_i64 ---> true
+        //  CAST(now() as int64) < cast(to_timestamp(...) as int64) + 50000_i64 ---> false
         let expr = cast_to_int64_expr(now_expr())
             .lt(cast_to_int64_expr(to_timestamp_expr(ts_string)) + lit(50000i64));
```

Comment:
maybe we can adjust the literal so it still evaluates to true so that we don't lose coverage 🤔